src/httpserver/http_request.hpp

```diff
@@ -442,7 +444,7 @@ class http_request
         **/
         void set_arg(const std::string& key, const std::string& value)
         {
-            this->args[key] = value;
+            this->args[key] = value.substr(0,content_size_limit);
```

Comment:
I did think of that. However, if I threw an exception, the functions `webserver::build_request_args`, and `webserver::post_iterator` would have to catch it immediately. They are invoked by libmicrohttpd, so they can not propagate exceptions.  Then we would still need a way to indicate that too many characters were POST'ed.

With the solution I gave, the user can check whether each of the args has size >= content_size_limit. We could even add a simple function to check overflow.  A C++98 version would be something like.

```c++
bool overflow() const
{
    for (std::map<std::string, std::string, http::arg_comparator>::const_iterator arg=args.begin();
          arg!=args.end(); ++arg)
    {
        if (arg->second.size()>=content_size_limit)
          return true;
    }
    return false;
}
```

Let me know what you think.